packages/react/src/utils.ts

```diff
@@ -0,0 +1,173 @@
+ * Not intended for creating outside of Gadget-owned code.
+ **/
+export class ErrorWrapper extends Error {
+  /** @private */
+  static forClientSideError(error: Error, response?: any) {
+    return new ErrorWrapper({
+      executionErrors: [error],
+      response,
+    });
+  }
+  /** @private */
+  static forErrorsResponse(errors: Record<string, any>[], response?: any) {
+    return new ErrorWrapper({
+      executionErrors: errors.map(gadgetErrorFor),
+      response,
+    });
+  }
+  /** @private */
+  static forMaybeCombinedError(error?: CombinedError | null) {
+    if (!error) return undefined;
+    return new ErrorWrapper({
+      networkError: error.networkError,
+      executionErrors: error.graphQLErrors,
+      response: error.response,
+    });
+  }
+
+  /** @private */
+  static ensureData(result: UseQueryState<any>, dataPath: string[]) {
+    const nonNullableError = getNonNullableError(result, dataPath);
+    let error = ErrorWrapper.forMaybeCombinedError(result.error);
+    if (!error && nonNullableError) {
+      error = ErrorWrapper.forClientSideError(nonNullableError);
+    }
+    return error;
+  }
+
+  /** Class name of this error -- always `ErrorWrapper` */
+  public name: "ErrorWrapper";
+  /** Error message for this error. Derived from the other errors this wraps. */
+  public message: string;
+  /**
+   * A list of errors encountered by the backend when processing a request. Populated if the client successfully communicated with the backend, but the backend was unable to process the request and rejected it with an error.
+   * Includes GraphQL syntax errors, missing or invalid argument errors, data validation errors, or unexpected errors encountered when running backend logic.
+   **/
+  public executionErrors: (GraphQLError | GadgetError)[];
+  /**
+   * An error encountered when trying to communicate with the backend from the client. Includes things like connection timeouts, connection interrupts, or no internet connection errors
+   **/
+  public networkError?: Error;
+  /**
+   * A list of errors the backend reported for specific fields being invalid for the records touched by an action. Is a shortcut for accessing the validation errors of a `GadgetInvalidRecordError` if that's what is in the `executionErrors`.
+   **/
+  public validationErrors?: InvalidFieldError[];
+  /**
+   * A list of errors encountered by the backend when processing a request. Populated if the client successfully communicated with the backend, but the backend was unable to process the request and rejected it with an error.
+   * Includes GraphQL syntax errors, missing or invalid argument errors, data validation errors, or unexpected errors encountered when running backend logic.
+   *
+   * This property allows this object to match the interface of urql's `CombinedError` object.
+   *
+   * @deprecated
```

Comment:
Not sure if there's a small message we could include with `@deprecated` to say what to use now, but I always like it when a lib specifies it :)